code/HUXt.py

```diff
@@ -1576,3 +1732,241 @@ def _setup_dirs_():
                 print('Error, invalid path, check config.dat: ' + val)
 
     return dirs
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+####################################################################################
+def mapVinwards(vouter = np.ones(128)*400*(u.km/u.s), router = 30*u.solRad, rinner = 10*u.solRad):
+    """
+    Function to map a longitudinal V  series from router (in rs) to rinner (in rs)
+    """
+    
+    #get the acceleration parameters
+    constants= huxt_constants()
+    alpha = constants['alpha']  # Scale parameter for residual SW acceleration
+    rH = constants['r_accel'].to(u.kilometer).value  # Spatial scale parameter for residual SW acceleration
+    Tsyn=constants['synodic_period'].to(u.s).value
+      
+    #compute the longitude grid from the length of the vouter input variable
+    lon, dlon, Nlon = longitude_grid(len(vouter))
+    
+    #convert to km
+    rout=router.to(u.kilometer).value
+    rin=rinner.to(u.kilometer).value
+    
+    #compute the speed at the new inner boundary height (using Vacc term, equation 5 in the paper)
+    v0=vouter.value/(1 + alpha*(1-np.exp((rin-rout)/rH)))
+
+    #compute the transit time from the new to old inner boundary heights (i.e., integrate equations 3 and 4 wrt to r)
+    A = v0 + alpha*v0
+    T_integral = rH *np.log (A*np.exp(rout/rH) - alpha*v0*np.exp(rin/rH) )/A 
+    -  rH*np.log (A* np.exp(rin/rH) - alpha*v0*np.exp(rin/rH) )/A
+
+    #work out the longitudinal shift
+    phis_new=_zerototwopi_(lon.value - (T_integral/Tsyn)*2*np.pi)
+
+    #regrid to regular longitudinal spacing (pad to make boundaries periodic)
+    phis_new_pad=np.concatenate((phis_new-2*np.pi,phis_new,phis_new+2*np.pi))
+    v0_pad=np.concatenate((v0,v0,v0))
+    vinner=np.interp(lon.value,phis_new_pad,v0_pad)
```

Comment:
I don't think we need to pad the array here. Numpy intep can do interpolation on data with periodic x values by specifying they period keyword with period=2*pi